src/parser.rs

```diff
@@ -407,6 +411,243 @@ pub fn parse_gcov(gcov_path: &Path) -> Result<Vec<(String,CovResult)>, ParserErr
+                        let class = fq_class.split('/').last().unwrap();                // Person$Age
+                        let top_class = class.clone().split("$").nth(0).unwrap();       // Person
+
+                        // Process all <method /> and <counter /> for this class
+                        let functions = parse_java_class(parser, class).unwrap();
+
+                        match results_hash.entry(top_class.to_string()) {
+                            hash_map::Entry::Occupied(obj) => {
+                                obj.into_mut().functions.extend(functions);
+                            },
+                            hash_map::Entry::Vacant(v) => {
+                                v.insert(CovResult {
+                                    functions,
+                                    lines: BTreeMap::new(),
+                                    branches: BTreeMap::new(),
+                                });
+                            },
+                        };
+                    },
+                    "sourcefile" => {
+                        // Find source file name
+                        let sourcefile = get_xml_attribute(&attributes, "name").unwrap(); // Person.java
+                        let class = sourcefile.trim_right_matches(".java"); // Person
+
+                        // Process all <line /> and <counter /> for this class
+                        let (lines, branches) = parse_java_sourcefile(parser).unwrap();
+
+                        // Save the result
+                        match results_hash.entry(class.to_string()) {
+                            hash_map::Entry::Occupied(obj) => {
+                                let obj = obj.into_mut();
+                                obj.lines = lines;
+                                obj.branches = branches;
+                            },
+                            hash_map::Entry::Vacant(v) => {
+                                v.insert(CovResult {
+                                    functions: HashMap::new(),
+                                    lines,
+                                    branches,
+                                });
+                            },
+                        };
+                    },
+                    &_ => {},
+                }
+            },
+            Ok(XmlEvent::EndElement{name}) => {
+                match name.local_name.as_str() {
+                    "package" => break,
+                    &_ => {}
+                }
+            },
+            Err(e) => {
+                return Err(ParserError::Parse(e.to_string()))
+            },
+            _ => {}
+        }
+    }
+
+    Ok(
+        results_hash.into_iter().filter(
+            |(_, result)| result.lines.len() > 0 || result.branches.len() > 0
+        ).map(
+            |(class, value)| (format!("{}/{}.java", package, class).trim_left_matches('/').to_string(), value)
```

Comment:
What are the initial '/' characters?

Reply:
There's at most one leading `/`, if the package is the empty string. The format will turn those into "/hello.java" instead of "hello.java".

Reply:
OK!

Reply:
Please add a comment